src/Symfony/Component/Console/Application.php

```diff
@@ -272,6 +281,22 @@ public function setCatchExceptions($boolean)
     }
 
     /**
+     * @return bool Whether errors are caught or not during commands execution
+     */
+    public function areErrorsCaught()
+    {
+        return $this->catchErrors;
+    }
+
+    /**
+     * @param bool $boolean Whether to catch errors or not during commands execution
+     */
+    public function setCatchErrors($boolean)
+    {
+        $this->catchErrors = (bool) $boolean;
+    }
```

Comment:
I doubt about the usefulness of `areErrorsCaught` right now, but this echoes `areExceptionsCaught`.
`setCatchErrors` is the way to tell the application instance to catch php 7 errors and cannot be removed.

IMHO, we still have to differentiate exceptions from errors, as the last ones are currently handled in the fullstack framework by the handler registered through the Debug component and the `DebugHandlersListener` from HttpKernel, whereas classic exceptions are handled by the Console component itself. Thus, relying on `catchExceptions` only will change the behavior using the full stack, which can be annoying and a BC break. The handler used in full stack does extra work on the exception. For instance:

Using full stack:

>  [Symfony\Component\Debug\Exception\UndefinedMethodException]                                                                  
  Attempted to call an undefined method named "Option" of class "Symfony\Component\Console\Input\ArgvInput".                    
  Did you mean to call e.g. "getOption", "getOptions", "getParameterOption", "hasOption", "hasParameterOption" or "setOption"?

versus the console component standalone, with the `catchErrors` flag:

> [Error]                                                                       
  Call to undefined method Symfony\Component\Console\Input\ArgvInput::Option()